app/controllers/application_controller.rb

```diff
@@ -5,6 +5,142 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
   end
 
-end
\ No newline at end of file
+  get '/' do
+    if is_logged_in
+      redirect '/tweets'
+    else
+      erb :homepage
+    end
+  end
+
+  get '/signup' do
+    if is_logged_in
+      redirect '/tweets'
+    else
+      erb :"tweets/users/create_user"
+    end
+  end
+
+  post '/signup' do
+      if params[:username] != "" && params[:email] != "" && params[:password] != ""
+        @user = User.create(username: params[:username], email: params[:email], password: params[:password])
+        session[:user_id] = @user.id
+        redirect '/tweets'
+      else
+        redirect '/signup'
+      end
+  end
+
+  get '/login' do
+    if is_logged_in
+      redirect "/tweets"
+    else
+      erb :"tweets/users/login"
+    end
+  end
+
+  post '/login' do
+    @user = User.find_by(username: params[:username])
+    if @user && @user.authenticate(params[:password])
+       session[:user_id] = @user.id
+       redirect "/tweets"
+     else
+       erb :"tweets/users/login", locals: {message: "Invalid username or password! Please try again."}
+     end
+  end
+
+  get "/logout" do
+    if is_logged_in
+      session.clear
+      redirect '/login'
+    else
+      redirect "/"
+    end
+  end
+
+  get '/tweets' do
+    if is_logged_in
+      @user = current_user
+      erb :"tweets/tweets"
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/tweets/new' do
+    if is_logged_in
+      erb :'/tweets/create_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+  post '/tweets' do
+    if params["content"] != ""
+      @tweet = Tweet.create(content: params["content"], user_id: session[:user_id])
```

Comment:
Another way to do this is to make use of the `has_many` relationship on User:
```
@tweet = current_user.tweets.create(content: params[:content])
```
Both approaches are good, but in some cases you may find this option useful.